core/src/main/java/com/graphhopper/util/exceptions/GHIllegalArgumentException.java

```diff
@@ -0,0 +1,31 @@
+package com.graphhopper.util.exceptions;
+
+import java.util.Map;
+
+/**
+ *
+ * Extends IllegalArgumentException because we explicitly catch that in the GraphHopperServlet - TODO we might use something else?
+ *
+ * @author Robin Boldt
+ */
+public abstract class GHIllegalArgumentException extends IllegalArgumentException
```

Comment:
Instead of an abstract class I would prefer to have an interface. E.g.

interface GHException {
 Map<String, String>getDetails()
}

Then we can extend from IllegalArgumentException and implement GHException for the CannotFindPointException but can create other exceptions with such details easily.

WDYT?